master/buildbot/process/botmaster.py

```diff
@@ -28,7 +28,53 @@
 from buildbot.util import service
 
 
-class BotMaster(service.ReconfigurableServiceMixin, service.AsyncMultiService):
+class LockRetrieverMixin:
+
+    @defer.inlineCallbacks
+    def getLockByID(self, lockid, config_version):
+        ''' Convert a Lock identifier into an actual Lock instance.
+            @lockid: a locks.MasterLock or locks.WorkerLock instance
+            @config_version: The version of the config from which the list of locks has been
+                acquired by the downstream user.
+            @return: a locks.RealMasterLock or locks.RealWorkerLock instance
+
+            The real locks are tracked using lock ID and config_version. The latter is used as a
+            way to track most recent properties of real locks.
+
+            This approach is needed because there's no central registry of lock access instances
+            that are used within a Buildbot configuration. All lock accesses bring all lock
```

Comment:
Ah.. I now understand your point

```suggestion
            that are used within a Buildbot master.cfg (like there is for e.g c['builders']). All lock accesses bring all lock
```

Reply:
Fixed now.